tests/components/vesync/test_diagnostics.py

```diff
@@ -0,0 +1,117 @@
+"""Tests for the diagnostics data provided by the VeSync integration."""
+import json
+from unittest.mock import patch
+
+from pyvesync.helpers import Helpers
+
+from homeassistant.components.vesync.const import DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.setup import async_setup_component
+
+from .common import (
+    call_api_side_effect__no_devices,
+    call_api_side_effect__single_fan,
+    call_api_side_effect__single_humidifier,
+)
+
+from tests.common import load_fixture
+from tests.components.diagnostics import (
+    get_diagnostics_for_config_entry,
+    get_diagnostics_for_device,
+)
+from tests.typing import ClientSessionGenerator
+
+
+async def test_async_get_config_entry_diagnostics__no_devices(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+    config_entry: ConfigEntry,
+    config: ConfigType,
+) -> None:
+    """Test diagnostics for config entry."""
+    with patch.object(Helpers, "call_api") as call_api:
+        call_api.side_effect = call_api_side_effect__no_devices
+        assert await async_setup_component(hass, DOMAIN, config)
+        await hass.async_block_till_done()
+
+    diag = await get_diagnostics_for_config_entry(hass, hass_client, config_entry)
+
+    assert isinstance(diag, dict)
+    assert diag == json.loads(
```

Comment:
You can use our new snapshot feature for this! 👍 

Docs: https://developers.home-assistant.io/docs/development_testing?_highlight=snapshot#snapshot-testing

Reply:
fixed to use snapshot